### ISSUE TITLE
[Inline Message] Prepare an explanation of why we don't use inline messages with actions and what we recommend using instead

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3266]
## Current Issue 

Teams using SLDS want to use Inline Messages with actions, which is not accessible. It take time to create custom message each time somebody is asking about it.

## Possible solution

Creating document on documentation website or Confluence that could be just pasted as answer would be an improvement.
This document should consist of:
1. explanation about why we don't recommend using inline message with actions
2. list of substitutes that can be used in that scenarios and still be accessible

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS

**Comment by a11ymiko** [comment]
### Conclusions from a11y cheampions meeting
Inline messages cannot include actions because:
1. Actions are not auto-focused and blind users may have issues searching for new contents to perform an action
2. Some browser + screen reader combination won't announce that inline message have actions as aria-live would announce only static text, not active components.

Alternatives for Inline message with actions:
1. Alert dialog / Dialog should be used if message and actions are so crucial that user workflow should be totally interrupted. https://www.w3.org/WAI/ARIA/apg/patterns/alertdialog/examples/alertdialog/
2. Error summary can be used in scenarios when user should stay at the same page. It important then to automatically move user's focus into error summary so actions are easy to find. https://service-manual.nhs.uk/design-system/components/error-summary

**Comment by a11ymiko** [comment]
## Next step:
Create a way to inform CFAs clearly about why Inline Messages cannot be used with actions regarding a11y and how to use Dialog with actions instead where needed.

If CFAs will still need component that behaves similarly to Inline Message but is used to list errors with skiplinks to each form field then adding Error Summary to SLDS would be a good idea.
